Source/FieldSolver/FiniteDifferenceSolver/FiniteDifferenceAlgorithms/SphericalYeeAlgorithm.H

```diff
@@ -0,0 +1,163 @@
+/* Copyright 2020 Remi Lehe
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_FINITE_DIFFERENCE_ALGORITHM_SPHERICAL_YEE_H_
+#define WARPX_FINITE_DIFFERENCE_ALGORITHM_SPHERICAL_YEE_H_
+
+#include "Utils/WarpXConst.H"
+
+#include <AMReX.H>
+#include <AMReX_Array4.H>
+#include <AMReX_Gpu.H>
+#include <AMReX_REAL.H>
+
+#include <array>
+#include <cmath>
+
+
+/**
+ * This struct contains only static functions to initialize the stencil coefficients
+ * and to compute finite-difference derivatives for the Spherical Yee algorithm.
+ */
+struct SphericalYeeAlgorithm {
+
+    static void InitializeStencilCoefficients (
+        std::array<amrex::Real,3>& cell_size,
+        amrex::Vector<amrex::Real>& stencil_coefs_r) {
+
+        using namespace amrex::literals;
+        // Store the inverse cell size along each direction in the coefficients
+        stencil_coefs_r.resize(1);
+        stencil_coefs_r[0] = 1._rt/cell_size[0];  // 1./dr
+    }
+
+    /** Compute the maximum, CFL-stable timestep
+     *
+     * Compute the maximum timestep, for which the scheme remains stable
+     * under the Courant-Friedrichs-Levy limit.
+     */
+    static amrex::Real ComputeMaxDt (amrex::Real const * const dx) {
+        using namespace amrex::literals;
+        // Check this value! This was determined empirically (for one case)
+        const amrex::Real alpha = 0.001_rt;
+        const amrex::Real delta_t = 1._rt / ( std::sqrt( (1._rt + alpha) / (dx[0]*dx[0])) * PhysConst::c );
+        return delta_t;
+    }
+
+    /**
+     * \brief Returns maximum number of guard cells required by the field-solve
+     */
+    static amrex::IntVect GetMaxGuardCell () {
+        // The spherical solver requires one guard cell in each dimension
+        return amrex::IntVect{AMREX_D_DECL(1,1,1)};
+    }
+
+    /** Applies the differential operator `1/r * d(rF)/dr`,
+     * where `F` is on a *nodal* grid in `r`
+     * and the differential operator is evaluated on a *cell-centered* grid.
+     * The input parameter `r` is given at the cell-centered position */
+    AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+    static amrex::Real UpwardDrr_over_r (
```

Comment:
What is the reason behind choosing `UpwardDrr_over_r` for this function's name (as well as similar subsequent functions)?

The name is using a combination of `PascalCase` and `snake_case`, and `Drr` to denote the operator `d(rF)/dr`.

Maybe we could try to use one style only and also make `Drr` more descriptive, e.g., `upward_radial_derivative_rF_over_r` or `upward_radial_grad_rF_over_r` or similar (although `F` may not be the best choice since it is related to div(E) cleaning).

I know this naming is used elsewhere, so this could a follow-up cleanup as well.

Reply:
This could be addressed in a separate PR, since it will also affect RZ.